spec.html

```diff
@@ -22277,6 +22277,194 @@ <h1>Example Source Text Module Record Graphs</h1>
+
+          <p>This abstract method performs the following steps:</p>
+
+          <emu-alg>
+            1. Let _module_ be this Dynamic Module Record.
+            1. Assert: _module_.[[Status]] is not `"instantiating"` or `"evaluating"`.
+            1. Let _realm_ be module.[[Realm]].
+            1. Assert: _realm_ is a valid Realm.
+            1. Let _env_ be NewModuleEnvironment(_realm_.[[GlobalEnv]]).
+            1. Set _module_.[[Environment]] to _env_.
+            1. Set _module_.[[Status]] to `"instantiated"`.
+            1. Return *undefined*.
+          </emu-alg>
+        </emu-clause>
+  
+        <emu-clause id="sec-moduleevaluation">
+          <h1>Evaluate ( ) Concrete Method</h1>
+
+          <p>The Evaluate concrete method of a Dynamic Module Record implements the corresponding Module Record abstract method.</p>
+          <p>Evaluate transitions this module's [[Status]] from `"instantiated"` to `"evaluated"`.</p>
+
+          <p>If execution results in an exception, that exception is recorded in the [[EvaluationError]] field and rethrown by future invocations of Evaluate.</p>
+
+          <p>Evaluation of dynamic modules calls out to _HostEvaluateDynamicModule_, before finalising the export names of the dynamic modules. Any uninitialized export bindings throw a reference error at this stage.</p>
+
+          <p>This abstract method performs the following steps:</p>
+
+          <emu-alg>
+            1. Let _m_ be this Dynamic Module Record.
+            1. Assert: _m_.[[Status]] is `"instantiated"` or `"evaluated"`.
+            1. Set _m_.[[Status]] to `"evaluating"`.
+            1. Let _result_ be HostEvaluateDynamicModule(_m_, _m_.[[ExportedBindings]]).
+            1. If _result_ is an abrupt completion, then
+              1. Assert: _m_.[[Status]] is `"evaluating"`.
+              1. Set _m_.[[Status]] to `"evaluated"`.
+              1. Set _m_.[[EvaluationError]] to _result_.
+              1. Return _result_.
+            1. Let _n_ be the value of _m_.[[Namespace]].
+            1. If _n_ is not _undefined_ then,
+              1. Assert _n_ is a Module Namespace Exotic Object.
+              1. Let _envRec_ be the value of _module_.[[Environment]].
+              1. For each string _exportName_ in _m_.[[ExportNames]], do
+                1. Assert: _envRec_ has a binding for _exportName_.
+                1. If the binding for _exportName_ in _envRec_ is uninitialized then,
+                  1. Let _error_ be a *ReferenceError* exception.
+                  1. Set _m_.[[Status]] to `"evaluated"`.
+                  1. Set _m_.[[EvaluationError]] to _error_.
+                  1. Return _error_.
+                1. If _n_.[[Exports]] does not contain a value for _exportName_ then,
+                  1. Add _exportName_ to _n_.[[Exports]] at the position corresponding to the sort order of `Array.prototype.sort` with *undefined* as _comparefn_.
```

Comment:
`1. Append _exportName_ to the end of _n_.[[Exports]].`

Reply:
I've updated this to use `Insert` based on "sort order".